include/seqan3/io/detail/output_iterator_conversion_adaptor.hpp

```diff
@@ -0,0 +1,217 @@
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ * \brief Output iterator adaptor that takes care of char to alphabet conversion in the assignment.
+ */
+
+#pragma once
+
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/core/concept/core.hpp>
+#include <seqan3/core/concept/iterator.hpp>
+#include <seqan3/io/concept.hpp>
+#include <seqan3/range/container/concept.hpp>
+
+#include <range/v3/range_traits.hpp>
+#include <seqan3/range/concept.hpp>
+
+#include <utility>
+
+namespace seqan3::detail
+{
+
+/*!\brief Output iterator adaptor that converts `char` into the given target type when assigning to it.
+ * \tparam oiter_type The adapted output_iterator.
+ * \tparam alpha_type The target alphabet type to convert to.
+ * \ingroup io
+ *
+ * \details
+ *
+ * Adapts any output iterator to a char convertible iterator, such that explicit conversions
+ * from char to the `alpha_type` is taken care of during assignment.
+ *
+ * ### Example
+ *
+ * ```cpp
+ * using insert_iter = ranges::back_insert_iterator<std::vector<dna4>>;
+ * using out_iter    = seqan3::detail::output_iterator_conversion_adaptor<insert_iter, dna4>;
+ *
+ * std::vector<dna4> out_vec;
+ * out_iter it{insert_iter{out_vec}};
+ *
+ * *it = 'A';
+ * *it = 'C';
+ * *it = 'G';
+ * *it = 'T';
+ *
+ * for (auto && c : out_vec)
+ *     std::cout << to_char(c);
+ * std::cout << '\n'; // prints ACGT
+ * ```
+ */
+template <typename oiter_type, typename alpha_type = char>
+//!\cond
+    requires alphabet_concept<alpha_type> && output_iterator_concept<oiter_type, alpha_type> &&
+        requires (alpha_type t)
+        {
+            { assign_char(t, 'a') }            -> alpha_type;
+            { assign_char(std::move(t), 'a') } -> alpha_type &&;
+        }
+//!\endcond
+class output_iterator_conversion_adaptor
+{
+private:
+    //!\brief The adapted output iterator.
+    oiter_type oiter;
+
+public:
+
+    /*!\name Member types
+     * \{
+     */
```

Comment:
`\brief Associated types are void for output iterators, see also LINK`

So, other people reading it, know whats going on.